packages/salesforcedx-vscode-core/src/util/orgUtil.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { Aliases } from '@salesforce/core';
+import { OUTPUT_CHANNEL } from '../channels';
+import { nls } from '../messages';
+import { notificationService, ProgressNotification } from '../notifications';
+import { FileInfo, OrgList } from '../orgPicker';
+
+export async function setUpOrgExpirationWatcher() {
+  // Run once to start off with.
+  await checkForExpiredOrgs();
+
+  /*
+  Comment this out for now.  For now, we are only going to check once on activation,
+  however it would be helpful if we also checked once a day.  If we decide to also
+  check once a day, uncomment the following code.
+
+  // And run again once every 24 hours.
+  setInterval(async () => {
+    await checkForExpiredOrgs();
+  }, 1000 * 60 * 60 * 24);
+  */
+}
+
+export async function checkForExpiredOrgs() {
+  try {
+    const daysBeforeExpire = 5;
+    const today = new Date();
+    const fiveDaysFromNow = new Date();
```

Comment:
For flexibility, might be better to rename to something like `daysUntilExpiration`?

Reply:
done